### ISSUE TITLE
OCaml 5.2 support

### ISSUE THREAD

**Issue by kit-ty-kate** [ocsigen/eliom#785]
eliom currently fails with the following error message on OCaml 5.2:
```
#=== ERROR while compiling eliom.10.3.1 =======================================#
# context              2.2.0~beta2~dev | linux/x86_64 | ocaml-variants.5.2.0+trunk | file:///home/opam/opam-repository
# path                 ~/.opam/5.2/.opam-switch/build/eliom.10.3.1
# command              ~/.opam/5.2/bin/dune build -p eliom -j 1 @install
# exit-code            1
# env-file             ~/.opam/log/eliom-19-bb1508.env
# output-file          ~/.opam/log/eliom-19-bb1508.out
### output ###
# (cd _build/default && /home/opam/.opam/5.2/bin/ocamlopt.opt -w -40 -g -I src/ppx/.ppx_utils.objs/byte -I src/ppx/.ppx_utils.objs/native -I /home/opam/.opam/5.2/lib/ocaml-compiler-libs/common -I /home/opam/.opam/5.2/lib/ocaml-compiler-libs/shadow -I /home/opam/.opam/5.2/lib/ocaml/compiler-libs -I /home/opam/.opam/5.2/lib/ppx_derivers -I /home/opam/.opam/5.2/lib/ppxlib -I /home/opam/.opam/5.2/lib/ppxlib/ast -I /home/opam/.opam/5.2/lib/ppxlib/astlib -I /home/opam/.opam/5.2/lib/ppxlib/print_diff -I /home/opam/.opam/5.2/lib/ppxlib/stdppx -I /home/opam/.opam/5.2/lib/ppxlib/traverse_builtins -I /home/opam/.opam/5.2/lib/sexplib0 -I /home/opam/.opam/5.2/lib/stdlib-shims -intf-suffix .ml -no-alias-deps -o src/ppx/.ppx_utils.objs/native/ppx_eliom_utils.cmx -c -impl src/ppx/ppx_eliom_utils.pp.ml)
# File "src/ppx/ppx_eliom_utils.ml", line 355, characters 37-40:
# 355 |           Typ.arrow (label_of_string lab) (type_of_out_type ty1)
#                                            ^^^
# Error: This expression has type "Asttypes.arg_label"
#        but an expression was expected of type "string"
# (cd _build/default && /home/opam/.opam/5.2/bin/ocamlc.opt -w -40 -g -bin-annot -I src/ppx/.ppx_utils.objs/byte -I /home/opam/.opam/5.2/lib/ocaml-compiler-libs/common -I /home/opam/.opam/5.2/lib/ocaml-compiler-libs/shadow -I /home/opam/.opam/5.2/lib/ocaml/compiler-libs -I /home/opam/.opam/5.2/lib/ppx_derivers -I /home/opam/.opam/5.2/lib/ppxlib -I /home/opam/.opam/5.2/lib/ppxlib/ast -I /home/opam/.opam/5.2/lib/ppxlib/astlib -I /home/opam/.opam/5.2/lib/ppxlib/print_diff -I /home/opam/.opam/5.2/lib/ppxlib/stdppx -I /home/opam/.opam/5.2/lib/ppxlib/traverse_builtins -I /home/opam/.opam/5.2/lib/sexplib0 -I /home/opam/.opam/5.2/lib/stdlib-shims -intf-suffix .ml -no-alias-deps -o src/ppx/.ppx_utils.objs/byte/ppx_eliom_utils.cmo -c -impl src/ppx/ppx_eliom_utils.pp.ml)
# File "src/ppx/ppx_eliom_utils.ml", line 355, characters 37-40:
# 355 |           Typ.arrow (label_of_string lab) (type_of_out_type ty1)
#                                            ^^^
# Error: This expression has type "Asttypes.arg_label"
#        but an expression was expected of type "string"
```

**Comment by balat** [comment]
Thanks!

**Comment by balat** [comment]
We probably need to wait for an update of ppxlib to fix that

**Comment by kit-ty-kate** [comment]
Update: beta1 got released today

**Comment by balat** [comment]
INcluded in 10.4.1
